apps/firebase-api/src/app/users.ts

```diff
@@ -0,0 +1,56 @@
+import admin from 'firebase-admin';
+import { logger } from 'firebase-functions';
+
+const getUserRef = (userId) =>
+  admin.firestore().collection('users').doc(userId);
+
+const addUNumToAllUsers = () => {
+  let uNum = 1;
+
+  const iterateAllUsers = async (nextPageToken?) => {
+    // List batch of users, 1000 at a time.
+    try {
+      const listUsersResult = await admin.auth().listUsers(1, nextPageToken);
+      const batch = admin.firestore().batch();
+
+      listUsersResult.users.forEach((userRecord) => {
+        uNum++;
+        batch.set(getUserRef(userRecord.uid), { uNum }, { merge: true });
+      });
+
+      await batch.commit();
+
+      if (listUsersResult.pageToken) {
+        // List next batch of users.
+        iterateAllUsers(listUsersResult.pageToken);
+      }
+    } catch (error) {
+      logger.error('Error listing users:', error);
+    }
+  };
+
+  iterateAllUsers();
+};
+
+export const addUniqueNumberToUser = async (user) => {
+  const userWithMaxUNumSnapshot = await admin
```

Comment:
I don't really like this actually. I'd rather have this function not run this check. Here's what should happen:

1. We should have the function `addUNumToAllUsers` deployed and we run it ONCE and only once for the organization.
2. We have a separate function that runs whenever someone new signs up.
3. Once the first function has run, we destroy it. At this point, we should be backpropigated and also have the second function running for all future users.